src/app/controls.tsx

```diff
@@ -13,226 +13,204 @@ import Slider from "@material-ui/core/Slider";
+import { MediaType, PlaybackMedia } from "./types";
 import Spotify from "spotify-web-api-js";
 import { useSpotifyToken } from "./util/auth";
 import { useDebounce } from "use-debounce";
 import useTrait from "./hooks/useTrait";
+import { SpotifyPlayer } from "./players/spotify-player";
 
 const useStyles = makeStyles((theme) => ({
-  text: {
-    padding: theme.spacing(2, 2, 0),
-  },
-  paper: {
-    paddingBottom: 50,
-  },
-  list: {
-    marginBottom: theme.spacing(2),
-  },
-  subheader: {
-    backgroundColor: "#6E7480",
-  },
-  appBar: {
-    top: "auto",
-    bottom: 0,
-    backgroundColor: "#6E7480",
-  },
-  grow: {
-    flexGrow: 1,
-  },
-  fabButton: {
-    position: "absolute",
-    zIndex: 1,
-    top: -30,
-    left: 0,
-    right: 0,
-    margin: "0 auto",
-  },
-  imgIcon: {
-    height: 50,
-    marginRight: theme.spacing(2),
-  },
-  volume: {
-    width: 200,
-    color: "#FFFFFF",
-  },
-  volumeSlider: {
-    color: "#FFFFFF",
-  },
-  playbackTimeSlider: {
-    width: 500,
-    color: "#FFFFFF",
-  },
+	text: {
+		padding: theme.spacing(2, 2, 0),
+	},
+	paper: {
+		paddingBottom: 50,
+	},
+	list: {
+		marginBottom: theme.spacing(2),
+	},
+	subheader: {
+		backgroundColor: "#6E7480",
+	},
+	appBar: {
+		top: "auto",
+		bottom: 0,
+		backgroundColor: "#6E7480",
+	},
+	grow: {
+		flexGrow: 1,
+	},
+	fabButton: {
+		position: "absolute",
+		zIndex: 1,
+		top: -30,
+		left: 0,
+		right: 0,
+		margin: "0 auto",
+	},
+	imgIcon: {
+		height: 50,
+		marginRight: theme.spacing(2),
+	},
+	volume: {
+		width: 200,
+		color: "#FFFFFF",
+	},
+	volumeSlider: {
+		color: "#FFFFFF",
+	},
+	playbackTimeSlider: {
+		width: 500,
+		color: "#FFFFFF",
+	},
 }));
 
 interface Props {
-  media: PlaybackMedia | undefined;
-  deviceID: string | undefined;
+	media: PlaybackMedia | undefined;
+	deviceID: string | undefined;
 }
 
 function Controls({ media, deviceID }: Props) {
-  const classes = useStyles();
-  const [timer, setTimer] = useState<any>();
-  const [volume, setVolume] = useState<number>(30);
-  const playbackTime = useTrait(0);
-  const [isPlaying, setIsPlaying] = useState(false);
-  const [debouncedVolume] = useDebounce(volume, 350, { maxWait: 2 });
+	const classes = useStyles();
+  	const [volume, setVolume] = useState<number>(30);
+	const [isPlaying, setIsPlaying] = useState<boolean>(false);
+  	const playbackTime = useTrait(0);
+	const [timer, setTimer] = useState<any>();
+	const [debouncedVolume] = useDebounce(volume, 350, { maxWait: 2 });
 
-  useEffect(() => {
-    console.log("This is called when song is clicked on in song browser.");
-    if (media) {
-      useSpotifyToken().then((token) => {
-        var spotifyApi = new Spotify();
-        spotifyApi.setAccessToken(token);
-        spotifyApi.play({ uris: [media.id], device_id: deviceID });
-        setIsPlaying(true);
-        updateProgressBar(0);
-      });
-    }
-  }, [media]);
+	const spotifyPlayer: SpotifyPlayer = new SpotifyPlayer(
```

Comment:
I think we want to encapsulate the spotify player in a generic PlayerHandler as well